draft-ietf-taps-impl.md

```diff
@@ -963,3 +963,46 @@ If the application indicates a bound on the expected Send or Receive bitrate, an
 
 * Cost Preferences:
 If the application indicates a preference to avoid expensive paths, and some paths are associated with a monetary cost, an implementation should decrease the ranking of such paths. If the application indicates that it prohibits using expensive paths, paths that are associated with a cost should be purged from the decision tree.
+
+# Reasons for errors {#appendix-reasons-errors}
+
+The Transport Services API {{I-D.ietf-taps-interface}} allows for the seven generic error types to specify a more detailed reason as to why an error occurred. This appendix lists some of the possible reasons.
+
+* BadConfiguration:
+The transport properties and endpoints provided by the application are either contradictory or incomplete. Examples include the lack of a remote endpoint on an active open or using a multicast group address while not requesting a unidirectional receive.
+
+* NoCandidates:
+None of the available transport protocols can satisfy the transport properties provided by the application.
+
+* Unresolvable:
+The remote or local specifier provided by the application can not be resolved.
+
+* CantEstablish:
+The TAPS system was unable to establish a transport-layer connection to the remote endpoint specified by the application.
+
+* PolicyProhibited:
+The system policy prevents the TAPS system from performing the action requested by the application. 
+
+* NotCloneable:
+The protocol stack is not capable of being cloned. 
+
+* MessageToLarge:
+The message size is too big for the taps system to handle.
+
+* ProtocolFailure:
+The underlying protocol stack failed. 
```

Comment:
What's the difference to "CantEstablish"?

Reply:
This one was intended to be used for protocol failures during receive and send. It might not be necessary to keep both as ProtocolFailure pretty much covers "CantEstablish".